lib/_http_outgoing.js

```diff
@@ -461,31 +458,30 @@ OutgoingMessage.prototype.write = function(chunk, encoding, callback) {
   // signal the user to keep writing.
   if (chunk.length === 0) return true;
 
+  // By corking the socket and uncorking after a tick, we manage to
+  // batch writes together, i.e. the header with the body.
+  if (this.connection && !this._corkedForWrite) {
+    this.connection.cork();
+    this._corkedForWrite = true;
+
+    process.nextTick(() => {
+      this.connection.uncork();
+      this._corkedForWrite = false;
+    });
```

Comment:
do not allocate a closure here, it will slow things down considerably. Use `connectionCorkNT` as it was done before.